src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java

```diff
@@ -757,6 +809,89 @@ public void reset() {
         return mDeviceInfo;
     }
 
+    /**
+     * Use this method to opt-out a user from tracking. Events and people updates that haven't been
+     * flushed yet will be deleted. Use {@link #flush()} before calling this method if you want
+     * to send all the queues to Mixpanel before.
```

Comment:
is it allowed to flush the events on the device after the user opt out in gdpr?

Reply:
if the user wants to send everything that has been tracked - yes.